include/swift/AST/PackExpansionMatcher.h

```diff
@@ -186,6 +186,23 @@ class TypeListPackMatcher {
       }
     }
 
+    // Both sides have equal number of pack expansion type.
+    if (lhsElts.size() == rhsElts.size()) {
```

Comment:
We discussed this offline but for posterity - the per-index matching is followed by adding a conversion constraint for each element pair to double-check their types which means that this example would fail (and it does).